pom.xml

```diff
@@ -1843,35 +1843,47 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
-                      <allowedImport>org.rocksdb.TransactionLogIterator.BatchResult</allowedImport>
-                      <allowedImport>org.rocksdb.TickerType</allowedImport>
-
-                      <!-- Allow RocksObjects whose native pointer is managed by RocksDB. -->
-                      <allowedImport>org.rocksdb.ColumnFamilyHandle</allowedImport>
-                      <allowedImport>org.rocksdb.Env</allowedImport>
-                      <allowedImport>org.rocksdb.Statistics</allowedImport>
-
-                      <!-- Allow RocksDB constants and static methods to be used. -->
-                      <allowedImport>org.rocksdb.RocksDB.*</allowedImport>
-                    </allowedImports>
-                    <exclusion>org.apache.hadoop.hdds.utils.db.managed.*</exclusion>
+                  <RestrictImports>	
+                    <includeTestCode>false</includeTestCode>	
+                    <reason>Use managed RocksObjects under org.apache.hadoop.hdds.utils.db.managed instead.</reason>	
+                    <!-- By default, ban all the classes in org.rocksdb -->	
+                    <bannedImport>org.rocksdb.**</bannedImport>	
+                    <allowedImports>	
+                      <!-- Allow non-RocksObject classes. -->	
+                      <allowedImport>org.rocksdb.BlockBasedTableConfig</allowedImport>	
+                      <allowedImport>org.rocksdb.ColumnFamilyDescriptor</allowedImport>	
+                      <allowedImport>org.rocksdb.CompactionStyle</allowedImport>	
+                      <allowedImport>org.rocksdb.HistogramData</allowedImport>	
+                      <allowedImport>org.rocksdb.HistogramType</allowedImport>	
+                      <allowedImport>org.rocksdb.Holder</allowedImport>	
+                      <allowedImport>org.rocksdb.InfoLogLevel</allowedImport>	
+                      <allowedImport>org.rocksdb.OptionsUtil</allowedImport>	
+                      <allowedImport>org.rocksdb.RocksDBException</allowedImport>	
+                      <allowedImport>org.rocksdb.StatsLevel</allowedImport>	
+                      <allowedImport>org.rocksdb.TransactionLogIterator.BatchResult</allowedImport>	
+                      <allowedImport>org.rocksdb.TickerType</allowedImport>	
+
+                      <allowedImport>org.rocksdb.AbstractEventListener</allowedImport>	
+                      <allowedImport>org.rocksdb.Checkpoint</allowedImport>	
+                      <allowedImport>org.rocksdb.CompactionJobInfo</allowedImport>	
+                      <allowedImport>org.rocksdb.CompressionType</allowedImport>	
+                      <allowedImport>org.rocksdb.DBOptions</allowedImport>	
+                      <allowedImport>org.rocksdb.FlushOptions</allowedImport>	
+                      <allowedImport>org.rocksdb.LiveFileMetaData</allowedImport>	
+                      <allowedImport>org.rocksdb.Options</allowedImport>	
+                      <allowedImport>org.rocksdb.RocksDB</allowedImport>	
```

Comment:
Allowing these (e.g. `RocksDB` and `Options`) seems to circumvent the whole point of banning these imports.

If these imports are absolutely necessary for the diff tool, please add override the restrictions only for that module.

Reply:
done. overridden in submodule's pom.xml instead

Reply:
`org.rocksdb.RocksDB` is only used in the UT.
For `org.rocksdb.Options` I think we will migrate to a managed one later.